test/extended/util/disruption/frontends/known_backends.go

```diff
@@ -4,55 +4,109 @@ import (
 	"context"
 
 	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
 	"github.com/openshift/origin/pkg/monitor"
 	"github.com/openshift/origin/pkg/monitor/backenddisruption"
 	"github.com/openshift/origin/test/extended/util/cluster"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/rest"
 	e2e "k8s.io/kubernetes/test/e2e/framework"
 )
 
+const (
+	oauthRouteNamespace = "openshift-authentication"
+	oauthRouteName      = "oauth-openshift"
+)
+
 func StartAllIngressMonitoring(ctx context.Context, m monitor.Recorder, clusterConfig *rest.Config) error {
-	if err := createOAuthRouteAvailableWithNewConnections().StartEndpointMonitoring(ctx, m, nil); err != nil {
+	// Ingress monitoring checks for oauth and console routes to monitor healthz endpoints. Check availability
+	// before setting up any monitors.
+	available, err := isRouteAvailable(ctx, clusterConfig, oauthRouteNamespace, oauthRouteName)
+	if err != nil {
 		return err
 	}
-	if err := createOAuthRouteAvailableWithConnectionReuse().StartEndpointMonitoring(ctx, m, nil); err != nil {
-		return err
+	if available {
```

Comment:
This test originally does not test for `isRouteAvailable` nor for `configAvailable`. To align with the original functionality one should test for both `isRouteAvailable` and `configAvailable` and only then create all the connections and other objects. Or, return error otherwise to fail earlier and avoid initializing half of the functionality.

Reply:
The routes used for monitoring belong to the console and the oauth api server, which are part of a regular OCP cluster. Since we don't have them in MicroShift this should be good enough to determine whether disruption tests should be created.
The one over the configAvailable comes from [this PR](https://github.com/openshift/origin/pull/27460), which introduces capability checking and a fallback if capability checking fails (which means we should still create the disruption monitor). Since we don't have the `clusterversion` in MicroShift we need to skip it.

Reply:
What I meant was:
```go
	// Ingress monitoring checks for oauth and console routes to monitor healthz endpoints. Check availability
	// before setting up any monitors.
	routeAvailable, err := isRouteAvailable(ctx, clusterConfig, oauthRouteNamespace, oauthRouteName)
	if err != nil {
		return err
	}
	configAvailable, err := isConfigAPIGroupAvailable(clusterConfig)
	if err != nil {
		return err
	}

	if !routeAvailable || !configAvailable {
		return fmt.Error("Route or config not available")
	}
	if routeAvailable {
		if err := createOAuthRouteAvailableWithNewConnections().StartEndpointMonitoring(ctx, m, nil); err != nil {
			return err
		}
		if err := createOAuthRouteAvailableWithConnectionReuse().StartEndpointMonitoring(ctx, m, nil); err != nil {
			return err
		}
	}
	if configAvailable {
		// Some jobs explicitly disable the console and other features. Check if it's disabled and if so,
		// do not run a disruption monitoring backend for it.
		configClient, err := configclient.NewForConfig(clusterConfig)
...
```

Reply:
The problem with that check is returning the error. This would prevent tests from running in MicroShift, as it fails to prepare before launching any of them.
If there are no routes for oauth and/or console then there is no point in setting anything up, as those services might not be installed/active, and this should mean no error.
It is a monitor device watching for any possible disruptions during the whole test run. If there are no services to watch, there cant be any disruptions, therefore the prepare phase should progress and setup nothing.

Reply:
> If there are no routes for oauth and/or console then there is no point in setting anything up, as those services might not be installed/active, and this should mean no error.

Valid point.

> The problem with that check is returning the error. This would prevent tests from running in MicroShift, as it fails to prepare before launching any of them.

I was under assumption running `StartEndpointMonitoring` requires existence of the route. Though, checking the code it seems it does not. `disruptionSampler.produceSamples` will produce an error over non-existing route connection check. Though, the monitor will keep running and collecting samples. Seems fair to skip running `StartEndpointMonitoring` for both cases when there's no OAuth route.